### ISSUE TITLE
'allow_unconfigured' defaults to 1

### ISSUE THREAD

**Issue by smortex** [choria-plugins/action-policy#5]
The action-policy plugin setup various configuration settings in the MCollective server, one of them being `allow_unconfigured` which defaults to 1 (enabled).

When this setting is enabled, actions on agents which do not have a corresponding policy files are accepted unconditionally.

However, the Choria documentation practically says the opposite:

> [Choria sets up the popular Action Policy based authorization and does so in a default deny mode which means by default, no-one can make any requests](https://choria.io/docs/configuration/aaa/#authorization)

This only affects agents without a policy file, and most (probably all) agents available ship with some policy.  This problem was discovered while writing a site-specific custom agent, and failing to provide such a policy file.

**Comment by smortex** [comment]
> This only affects agents without a policy file, and most (probably all) agents available ship with some policy.

After further investigation, the `choria_util` agent does not ship with a policy file, so will break with this change.

**Comment by ripienaar** [comment]
Good catch, I suspect if we add to https://github.com/choria-io/mcollective-choria/blob/master/module/choria/data/plugin.yaml

```
mcollective_choria::policy_default: allow
```

it should solve the problem for that one - or at least retain current behaviour, then I think we're ok to merge this.

What do you think?

**Comment by smortex** [comment]
I guess it would solve the problem yes.  What I am unsure about is the sensitivity of the information available: the returned `client_stats` contain for example `in_msgs` and `out_msgs` which may be used to have insights on how a system is used…

I do not see this information as particularly sensitive, but you have more knowledge than me about a potential "risk" in this area :-)

**Comment by ripienaar** [comment]
Yeah I don’t think knowing that is going to tell anyone much meaningful info :)

**Comment by smortex** [comment]
I was forgetting about this and hit it again today :-)

> Good catch, I suspect if we add to choria-io/mcollective-choria:module/choria/data/plugin.yaml@master
>
> ```
> mcollective_choria::policy_default: allow
> ```
>
> it should solve the problem for that one - or at least retain current behaviour, then I think we're ok to merge this.

I opened choria-io/mcollective-choria#571 that I failed to do previously.

**Comment by ripienaar** [comment]
Thanks, I think long term I want this to be deny but for now its fine.

**Comment by smortex** [comment]
Just to be sure: #4 (which fix this issue) is **not** merged?  Did you want to close this issue?

**Comment by ripienaar** [comment]
Hmm, ok right I probably lost track of whats happening here as its been a while :) so you're saying we should merge both and that gets us default deny yeah?

**Comment by smortex** [comment]
Yeah, currently, when no policy is configured, action-policy allows all actions by default.  Since the documentation says the opposite, this PR was opened to fix this issue.

The choria-io/mcollective-choria module did not set any policy, so actions that where previously allowed would become blocked by default with this PR is merged, so another PR was opened in this project to provide a policy (and continue to allow all actions).

Now that choria-io/mcollective-choria has a policy, the default when no policy is configured can be set to deny actions, to align with what the doc says.

**Comment by ripienaar** [comment]
So as things stand we're on allow all - if we merge #4 we become deny all.

For sure I need to send heads ups to the list and stuff before we flip this over.

**Comment by ripienaar** [comment]
I've notified the list and merged the other one
